### ISSUE TITLE
saturation bug (1.12.2, F:2705 TAN:3.0.1.119)

### ISSUE THREAD

**Issue by audenga** [Glitchfiend/ToughAsNails#527]
similar to reported thirt bugs, i disabled thirst cos i thought that was doing it (config file https://pastebin.com/0qF0iPNg) and its still consuming all my saturation without healing, happens on both single player and multi player, with and without thirst enabled

note, i am using lurms modpack from twitch with ex nihilo and tinkers tool leveling because im on the great wall world setting

**Comment by dannydjdk** [comment]
I've run into what may be the same bug which is effectively preventing a player from regenerating health. When saturated, the game attempts to use the saturation to heal the player, but something is blocking the healing. Even regen potions do not work. In my case, removing the Tough Expansion mod solved the problem. The bug seems to be there. If I can manage to isolate it, I'll report the bug there.

**Comment by kd8lvt** [comment]
In the ToughExpansion config there is a option called "ThirstHealthFix" which requires your thirst to be full for you to heal. However, all it does is the block healing event itself (thus preventing you from refilling hearts), rather than saying "no, you can't heal because you don't have the food" (aka the proper way of doing it) which confuses Minecraft, resulting in your saturation to go down (or the healing potion to be drunk, etc.) but not heal you.

Disabling this config option (which is enabled by default) fixes this problem.

TL;DR: There's a config option in ToughExpansion's config file that causes this behavior, disable it and it's fixed. This is not a TAN bug.

**Comment by KnightMiner** [comment]
Has this been reported to ToughExpansion? Would be helpful to link the issue here for future people with the issue

**Comment by Forstride** [comment]
Just so people are aware, Tough Expansion is not associated with us at all.  You need to make sure that issues are actually being caused by TAN alone when reporting them.
